deploy/setup-project.sh

```diff
@@ -22,34 +22,65 @@ readonly KUBEDEPLOY="${PKGDIR}/deploy/kubernetes"
 readonly BIND_ROLES="roles/compute.storageAdmin roles/iam.serviceAccountUser projects/${PROJECT}/roles/gcp_compute_persistent_disk_csi_driver_custom_role"
 readonly IAM_NAME="${GCE_PD_SA_NAME}@${PROJECT}.iam.gserviceaccount.com"
 
-# Create or Update Custom Role
-if gcloud iam roles describe gcp_compute_persistent_disk_csi_driver_custom_role --project "${PROJECT}";
-then
-  gcloud iam roles update gcp_compute_persistent_disk_csi_driver_custom_role --quiet \
-          --project "${PROJECT}"                                                     \
-          --file "${PKGDIR}/deploy/gcp-compute-persistent-disk-csi-driver-custom-role.yaml"
-else
-  gcloud iam roles create gcp_compute_persistent_disk_csi_driver_custom_role --quiet \
-    --project "${PROJECT}"                                                           \
-    --file "${PKGDIR}/deploy/gcp-compute-persistent-disk-csi-driver-custom-role.yaml"
+function create_new_sa()
+{
+	# Delete Service Account Key
+	if [ -f "${GCE_PD_SA_DIR}/cloud-sa.json" ]; then
+	  rm "${GCE_PD_SA_DIR}/cloud-sa.json"
+	fi
+	# Delete ALL EXISTING Bindings
+	gcloud projects get-iam-policy "${PROJECT}" --format json > "${PKGDIR}/deploy/iam.json"
+	sed -i "/serviceAccount:${IAM_NAME}/d" "${PKGDIR}/deploy/iam.json"
+	gcloud projects set-iam-policy "${PROJECT}" "${PKGDIR}/deploy/iam.json"
+	rm -f "${PKGDIR}/deploy/iam.json"
+	# Delete Service Account
+	gcloud iam service-accounts delete "${IAM_NAME}" --project "${PROJECT}" --quiet || true
+
+	# Create new Service Account and Keys
+	gcloud iam service-accounts create "${GCE_PD_SA_NAME}" --project "${PROJECT}"
+}
+
+# Check if SA exists
+CREATE_SA=true
+SA_JSON=$(gcloud iam service-accounts list --filter="name:$IAM_NAME" --format="json")
```

Comment:
could we just do a `get` instead of a `list and filter` since we're only looking for one?

Reply:
I personally prefer this as trying to describe (get) instead of list prints an error to stdout if it fails, but I can still change it if you prefer